src/settings.cpp

```diff
@@ -16,98 +24,385 @@ namespace openmc {
-    if (temp_str == "nearest") {
-      temperature_method = TEMPERATURE_NEAREST;
-    } else if (temp_str == "interpolation") {
-      temperature_method = TEMPERATURE_INTERPOLATION;
+
+  if (run_mode == RUN_MODE_EIGENVALUE || run_mode == RUN_MODE_FIXEDSOURCE) {
+    // Read run parameters
+    get_run_parameters(node_mode);
+
+    // Check number of active batches, inactive batches, and particles
+    if (n_batches <= n_inactive) {
+      fatal_error("Number of active batches must be greater than zero.");
+    } else if (n_inactive < 0) {
+      fatal_error("Number of inactive batches must be non-negative.");
+    } else if (n_particles <= 0) {
+      fatal_error("Number of particles must be greater than zero.");
+    }
+  }
+
+  // Copy random number seed if specified
+  if (check_for_node(root, "seed")) {
+    auto seed = std::stoll(get_node_value(root, "seed"));
+    openmc_set_seed(seed);
+  }
+
+  // Check for electron treatment
+  if (check_for_node(root, "electron_treatment")) {
+    auto temp_str = get_node_value(root, "electron_treatment", true, true);
+    if (temp_str == "led") {
+      electron_treatment = ELECTRON_LED;
+    } else if (temp_str == "ttb") {
+      electron_treatment = ELECTRON_TTB;
     } else {
-      fatal_error("Unknown temperature method: " + temp_str);
+      fatal_error("Unrecognized electron treatment: " + temp_str + ".");
     }
   }
-  if (check_for_node(*root, "temperature_tolerance")) {
-    temperature_tolerance = std::stod(get_node_value(*root, "temperature_tolerance"));
+
+  // Check for photon transport
+  if (check_for_node(root, "photon_transport")) {
+    photon_transport = get_node_value_bool(root, "photon_transport");
+
+    if (!run_CE && photon_transport) {
+      fatal_error("Photon transport is not currently supported in "
+        "multigroup mode");
+    }
   }
-  if (check_for_node(*root, "temperature_multipole")) {
-    temperature_multipole = get_node_value_bool(*root, "temperature_multipole");
+
+  // Number of bins for logarithmic grid
+  if (check_for_node(root, "log_grid_bins")) {
+    n_log_bins = std::stoi(get_node_value(root, "log_grid_bins"));
+    if (n_log_bins < 1) {
+      fatal_error("Number of bins for logarithmic grid must be greater "
+        "than zero.");
+    }
   }
-  if (check_for_node(*root, "temperature_range")) {
-    auto range = get_node_array<double>(*root, "temperature_range");
-    temperature_range[0] = range[0];
-    temperature_range[1] = range[1];
+
+  // Number of OpenMP threads
+  if (check_for_node(root, "threads")) {
+#ifdef _OPENMP
+    if (openmc_n_threads == 0) {
+      openmc_n_threads = std::stoi(get_node_value(root, "threads"));
+      if (openmc_n_threads < 1) {
+        std::stringstream msg;
+        msg << "Invalid number of threads: " << openmc_n_threads;
+        fatal_error(msg);
+      }
+      omp_set_num_threads(openmc_n_threads);
+    }
+#else
+    if (openmc_master) warning("Ignoring number of threads.");
```

Comment:
We could make this error message more helpful by saying that OpenMC was compiled without OpenMP